lib/hal_api/controller/cache.rb

```diff
@@ -12,8 +12,19 @@ def index_cache_path
     ).cache_key
   end
 
+  def current_time
+    Datetime.now
+  end
+
   def show_cache_path
-    show_resource.updated_at.utc.to_i
+    timestamp = show_resource.updated_at || show_resource.created_at
+    timestamp = if timestamp.nil?
+                  # skip the cache
+                  current_time
+                else
+                  timestamp
```

Comment:
I can't think of a better workaround - basically this will act like a random val cache buster, as you said in your comments, and that seems like the right default.